Separate ArcMap and ArcGIS Pro code #642

csmoore opened the pull request:
We are just switching the head branch we are merging PR #644 into (branch: **Pro2.6** vs. dev).

Implements #642
Original PR: https://github.com/Esri/coordinate-conversion-addin-dotnet/pull/644 (we will close this one unmerged after this PR is merged)

Original PR #644 comments from @sumitzarkar :

We have implemented code for  following points:
 1. Removing: ArcMapAddinCoordinateConversion project/dll
 2. Removing:CoordinateConversionLibrary project/dll
 3. Refactoring the Pro project to remove dependencies on former CoordinateConversionLibrary
   a. remove view inheritance
  b. remove no-longer-needed classes (super/abstract classes)
  c. remove mediator/dispatcher (ex: convert to RelayCommands)
  d. use Pro styling for Dark Mode support
5. As part of this if you notice any hard-coded UI strings that should be converted to Resources (for I18n) please convert these to resources

sumitzarkar committed:
Implemented code for gitHub ticket #642 Sprint1/2

Implemented following points:
 1. Removing: ArcMapAddinCoordinateConversion project/dll
 2. Removing:CoordinateConversionLibrary project/dll
 3. Refactoring the Pro project to remove dependencies on former CoordinateConversionLibrary
    a. remove view inheritance

Note: This build is work in progress and some build errors can be observed

sumitzarkar committed:
Implemented code for gitHub ticket #642 Sprint2/2

- Completed refactoring

sumitzarkar committed:
Implemented code for gitHub ticket #642 Sprint 2/2 Bug fixes

- Fixed crash found after last commit

sumitzarkar committed:
Implemented code for gitHub ticket #642 Sprint 2/2 - Pro Styles

Updated code to make Pro Style Changes.

sumitzarkar committed:
Implemented code for gitHub ticket #642 Sprint 2/2 - Localization & Pro Styles for Buttons

Implemented code for gitHub ticket #642 Sprint 2/2 - Localization & Pro Styles for Buttons

csmoore committed:
Add resource dictionary

sumitzarkar committed:
Merge pull request #645 from Esri/CT-Ticket642-Edits

Add resource dictionary

sumitzarkar committed:
Implemented code review comments for #642

Implemented following points:
 1. Instead of NotificationObject, use the Pro sdk provided ViewModelBase
 2. Add the dependencies in Config.daml

csmoore commented:
@sumitzarkar We have finished our review and testing, the only remaining issues to close and merge this PR is:

Combined duplicate Base Classes into one class/file:

1. TabBaseViewModel and ProTabBaseViewModel
2. OutputCoordinateViewModel and ProOutputCoordinateViewModel

sumitzarkar commented:
@csmoore 

>>Combined duplicate Base Classes into one class/file:
TabBaseViewModel and ProTabBaseViewModel
OutputCoordinateViewModel and ProOutputCoordinateViewModel

We are working on this.

csmoore commented:
@sumitzarkar -as we discussed I added [2 more dark mode issues to the list above](https://github.com/Esri/coordinate-conversion-addin-dotnet/pull/646#issuecomment-595203790)

csmoore commented on source/CoordinateConversion/ProAppCoordConversionModule/Models/CoordinateConversionLibraryConfig.cs:
```diff
@@ -25,7 +26,7 @@
 namespace ProAppCoordConversionModule.Models
 {
-    public class CoordinateConversionLibraryConfig : NotificationObject
+    public class CoordinateConversionLibraryConfig : ViewModelBase
```
We may have been a little unclear in our [original feedback](https://github.com/Esri/coordinate-conversion-addin-dotnet/pull/644#discussion_r386048612), [ViewModelBase](https://pro.arcgis.com/en/pro-app/sdk/api-reference/#topic10691.html) is used for view model classes and [PropertyChangedBase](https://pro.arcgis.com/en/pro-app/sdk/api-reference/#topic10646.html) is used for all other classes.

sumitzarkar committed:
Fixed Pro Style Issues for browse Window - #642

Implemented following github tickets:

 - #647 Browse window does not stay consistent with Dark Theme mode while opening
 - #648 Browse window does not stay consistent with Dark Theme mode while importing

csmoore commented:
@sumitzarkar We have finished our review and testing, the only remaining issues to close and merge this PR is:

Combined duplicate Base Classes into one class/file:

TabBaseViewModel and ProTabBaseViewModel
OutputCoordinateViewModel and ProOutputCoordinateViewModel
[NEW] A couple of Dark Mode Form issues:

#647
#648

csmoore commented:
✔️  These issues have been tested and closed (Dark Mode Form issues) : #647 , #648

So only remaining issues are : 

1. Combined duplicate Base Classes into one class/file:
TabBaseViewModel and ProTabBaseViewModel
OutputCoordinateViewModel and ProOutputCoordinateViewModel
2. Change a few base class objects from `ViewModelBase` to `PropertyChangedBase`  - comment here:  https://github.com/Esri/coordinate-conversion-addin-dotnet/pull/646/commits/eff8118d3ece199b7831cf2ff861ef7427fec30c#r389223919

sumitzarkar committed:
Implemented code for #642

Combined duplicate Base Classes into one class/file:
1.  TabBaseViewModel and ProTabBaseViewMode
2.  OutputCoordinateViewModel and ProOutputCoordinateViewModel

csmoore commented:
[Successfully tested](https://github.com/Esri/coordinate-conversion-addin-dotnet/issues/642#issuecomment-601932387) - merging
